components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -871,6 +964,58 @@ subroutine radiation_init(state)
                      sampling_seq='rad_lwsw', flag_xyfill=.true.)
       endif
 
+      if (do_SPA_optics) then !initialize SPA
+         write(iulog,*) 'initializing SPA, iam = ', iam
+         !find current_month
+         call get_curr_date(year,month,day,tod)
+         current_month = month
+         write(iulog,*) 'current month is',current_month
+         if (month==12) then
+            next_month = 1
+         else
+            next_month = month + 1
+         end if
+
+         allocate(aerosol_optical_property(pcols,pver,nlwbands,begchunk:endchunk))
+         call get_aerosol_optical_property_from_file(current_month,'AER_TAU_LW','lwband',nlwbands,aerosol_optical_property)
+         call pbuf_set_field(pbuf, aer_tau_bnd_lw_mon_1_idx,aerosol_optical_property)
+         deallocate(aerosol_optical_property)
```

Comment:
The use of allocate and deallocate in this chunk of code I think is a way to re-initialize and recycle the `aerosol_optical_property` array.  I think you can also just get away with setting the whole array to a value each time, like `aerosol_optical_property(:,:,:,:) = 0.0` or something along those lines.